Summarize Current Session choice

FadhlanR opened the pull request:
In this PR, I implemented the third option for the new session settings. I added a new command, `SummarizeSessionCommand`, which retrieves the summary of the currently open room by calling `SendRequestViaProxyCommand`, which then forwards the request to the OpenRouter endpoint. I also need to extract helpers from ai-bot package so it can be reused in the host.


https://github.com/user-attachments/assets/4689a1b1-561f-4d6f-9b88-41cadb3320cf

FadhlanR committed:
extract ai-bot-helpers

FadhlanR committed:
Summarize Current Session choice

FadhlanR committed:
remove whitelisted property

FadhlanR committed:
update pnpm lock

FadhlanR committed:
Merge branch 'main' into cs-8925-summarize-content-session

FadhlanR committed:
potential fix

FadhlanR committed:
fix matrix test

FadhlanR commented on packages/host/tests/acceptance/prerender-test.gts:
```diff
@@ -158,6 +158,10 @@ module('Acceptance | prerender', function (hooks) {
   });
 
+  hooks.afterEach(function () {
+    delete (globalThis as any).__lazilyLoadLinks;
```
Not related to this PR, but if we don't delete __lazilyLoadLinks, the subsequent tests could fail because they expect it not to exist or false.

lukemelia commented on packages/ai-bot/lib/matrix/response-publisher.ts:
```diff
@@ -1,15 +1,17 @@
   APP_BOXEL_CONTINUATION_OF_CONTENT_KEY,
   APP_BOXEL_HAS_CONTINUATION_CONTENT_KEY,
+  sendErrorEvent,
+  sendMessageEvent,
```
function names need updating if moving to this context: sendErrorMatrixEvent, sendMessageMatrixEvent

lukemelia commented on packages/ai-bot/lib/debug.ts:
```diff
@@ -1,28 +1,18 @@
+import * as Sentry from '@sentry/node';
+import type { MatrixEvent as DiscreteMatrixEvent } from 'https://cardstack.com/base/matrix-event';
 import {
+  getPromptParts,
+  isRecognisedDebugCommand,
```
Similarly, `getAiAssistantPromptParts` and `isRecognisedAiAssistantDebugCommand`

lukemelia commented on packages/ai-bot/lib/set-title.ts:
```diff
@@ -21,6 +14,13 @@ import type {
+  type OpenAIPromptMessage,
+  isCodePatchResultStatusApplied,
+  isCommandResultStatusApplied,
+  attachedCardsToMessage,
+  getRelevantCards,
```
rename

lukemelia commented on packages/ai-bot/tests/helpers/fake-matrix-client.ts:
```diff
@@ -24,27 +27,22 @@ export class FakeMatrixClient implements MatrixClient {
       return { chunk: [] };
     },
-  };
+  } as any;
```
Why `any` here? Seems like a step backwards.

lukemelia commented on packages/ai-bot/tests/helpers/fake-matrix-client.ts:
```diff
@@ -54,7 +52,7 @@ export class FakeMatrixClient implements MatrixClient {
     this.eventId++;
     return { event_id: messageEventId.toString() };
-  }
+  }) as any;
```
Same question

lukemelia commented on packages/ai-bot/tests/chat-titling-test.ts:
```diff
@@ -21,16 +21,16 @@ import {
   REPLACE_MARKER,
   SEARCH_MARKER,
   SEPARATOR_MARKER,
+  constructHistory,
```
`constructHistoryFromAiAssistantRoom`

lukemelia commented on packages/ai-bot/tests/prompt-construction-test.ts:
```diff
@@ -31,6 +24,11 @@ import { FakeMatrixClient } from './helpers/fake-matrix-client';
 import {
   type LooseCardResource,
   skillCardRef,
+  buildPromptForModel,
+  getPromptParts,
+  getRelevantCards,
+  getTools,
```
Rename

lukemelia commented on packages/ai-bot/main.ts:
```diff
@@ -15,22 +15,20 @@ import {
+  PromptParts,
+  isRecognisedDebugCommand,
+  getPromptParts,
+  isInDebugMode,
```
rename

lukemelia reviewed (changes requested):
In addition to the comments I left, I think we need some visual progress feedback in order to land this. It's too slow to leave the spinner up for that long. Please share in the design channel along with your suggestions to get a design treatment.

FadhlanR committed:
remove as any

FadhlanR committed:
import from runtime-common/ai

FadhlanR committed:
Implement loading state

FadhlanR committed:
rename to skip

FadhlanR commented:
@lukemelia I implemented a loading state and added a button to skip the summarizing process. I also applied the same loading state to copying file history, since that action also involves sending a message.

https://github.com/user-attachments/assets/305a95f5-43a3-4663-aedc-5bff1d9b4046

lukemelia commented on packages/ai-bot/tests/helpers/fake-matrix-client.ts:
```diff
@@ -28,22 +28,41 @@ export class FakeMatrixClient extends MatrixClient {
-    content: IContent,
+    eventTypeOrThreadId: string | null,
+    contentOrEventType: IContent | string,
+    txnIdOrContent?: string | IContent,
```
This is getting rough. Should we switch to a named params style API?

FadhlanR committed:
Merge branch 'main' into cs-8925-summarize-content-session

FadhlanR committed:
lint fix
